chemprop/v2/cli/train.py

```diff
@@ -32,41 +32,308 @@ class TrainSubcommand(Subcommand):
 
     @classmethod
     def add_args(cls, parser: ArgumentParser) -> ArgumentParser:
-        return add_args(parser)
+        parser = add_common_args(parser)
+        return add_train_args(parser)
 
     @classmethod
     def func(cls, args: Namespace):
         process_args(args)
+        validate_args(args)
         main(args)
 
+def add_common_args(parser: ArgumentParser) -> ArgumentParser:
+    parser.add_argument(
+        "--logdir",
+        nargs="?",
+        const="chemprop_logs",
+        help="runs will be logged to {logdir}/chemprop_{time}.log. If unspecified, will use 'save_dir'. If only the flag is given (i.e., '--logdir'), then will write to 'chemprop_logs'",
+    )
 
-def add_args(parser: ArgumentParser) -> ArgumentParser:
+    data_args = parser.add_argument_group("input data parsing args")
+    data_args.add_argument(
+        "-s",
+        "--smiles_columns",
+        type=list[str],
+        # to do: make sure default is coded correctly
+        help="List of names of the columns containing SMILES strings. By default, uses the first :code:`number_of_molecules` columns.",
+    )
+    data_args.add_argument(
+        "--number_of_molecules",
+        type=int,
+        default=1,
+        help="Number of molecules in each input to the model. This must equal the length of :code:`smiles_columns` (if not :code:`None`).",
+    )
```

Comment:
can't we just guess this from `smiles-columns`?

Reply:
Yes and v1 does that if number of molecules isn't specified. Users can use `--number_of_molecules` to specify how many columns (starting with the first) contain SMILES in the input. If the user wants more flexibility in specifying non consecutive columns, they can use the `--smiles_columns` options instead. 

I think the CLI will be simpler without this option, but maybe we should ask MLPDS before making major changes to the CLI as I'm not sure that was in the scope of the v1 -> v2 transition.